java/statistical-query/src/main/java/com/esri/arcgisruntime/sample/statisticalquery/MainActivity.java

```diff
@@ -0,0 +1,171 @@
+    StatisticDefinition statDefinitionCount = new StatisticDefinition("POP", StatisticType.COUNT, "CityCount");
+
+    // add the statistics definitions to a list
+    List<StatisticDefinition> statDefinitions = new ArrayList<>();
+    statDefinitions.add(statDefinitionAvgPop);
+    statDefinitions.add(statDefinitionCount);
+    statDefinitions.add(statDefinitionMinPop);
+    statDefinitions.add(statDefinitionMaxPop);
+    statDefinitions.add(statDefinitionSumPop);
+    statDefinitions.add(statDefinitionStdDevPop);
+    statDefinitions.add(statDefinitionVarPop);
+
+    // create the statistics query parameters, pass in the list of definitions
+    StatisticsQueryParameters statQueryParams = new StatisticsQueryParameters(statDefinitions);
+
+    // if only using features in the current extent, set up the spatial filter for the statistics query parameters
+    if (mCurrentExtentCheckbox.isChecked()) {
+      // get the current extent (as an envelope) from the map view
+      Envelope currentExtent = (Envelope) mMapView.getCurrentViewpoint(Viewpoint.Type.BOUNDING_GEOMETRY)
+          .getTargetGeometry();
+
+      // set the statistics query parameters geometry with the envelope
+      statQueryParams.setGeometry(currentExtent);
+
+      // set the spatial relationship to Intersects (which is the default)
+      statQueryParams.setSpatialRelationship(QueryParameters.SpatialRelationship.INTERSECTS);
+    }
+
+    // set up an attribute filter
+    if (mGreater5mCheckbox.isChecked()) {
+      // set a where clause to get only cities with populations over five million
+      statQueryParams.setWhereClause("POP_CLASS = '5,000,000 and greater'");
+    }
+
+    // execute the statistical query with these parameters and await the results
+    ListenableFuture<StatisticsQueryResult> statQueryResultFuture = mWorldCitiesTable
+        .queryStatisticsAsync(statQueryParams);
+    statQueryResultFuture.addDoneListener(() -> {
+      try {
+        // get the result
+        StatisticsQueryResult statQueryResult = statQueryResultFuture.get();
+
+        // get iterator from the result
+        Iterator<StatisticRecord> statisticRecordIterator = statQueryResult.iterator();
+
+        // build a result string for display
+        StringBuilder result = new StringBuilder();
+        while (statisticRecordIterator.hasNext()) {
+          Map<String, Object> statisticsMap = statisticRecordIterator.next().getStatistics();
+          for (Map.Entry<String, Object> stat : statisticsMap.entrySet()) {
+            result.append(stat.getKey()).append(": ").append(String.format(Locale.US, "%,.0f", (Double) stat.getValue()))
+                .append("\n");
+          }
+        }
+
+        // show the results in a snackbar
+        Snackbar reportSnackbar = Snackbar.make(findViewById(R.id.activityMain), result, Snackbar.LENGTH_INDEFINITE);
+        reportSnackbar.setAction("Dismiss", view -> reportSnackbar.dismiss());
```

Comment:
Strictly speaking, material design guidelines say don't use a Snackbar with Dismiss or Cancel as the action text. Could change to 'New query' or 'Change query'?